tests/console/perl_bootloader.pm

```diff
@@ -15,19 +15,21 @@ use warnings;
 use utils;
 use package_utils;
 use power_action_utils 'power_action';
-use version_utils qw(is_sle check_version is_transactional);
+use version_utils qw(is_sle check_version is_transactional is_tumbleweed is_microos is_sle_micro);
 use transactional;
 
 sub run {
     my ($self) = @_;
+    # https://progress.opensuse.org/issues/165686
+    my $package = (is_tumbleweed() || is_sle_micro() || is_microos()) ? 'update-bootloader' : 'perl-Bootloader';
```

Comment:
Our tests should always consider openSUSE Tumbleweed as the default product without a version. That means any differing behaviour should have an explicit exclude rule for the "older products" and potentially an explanation why the behaviour should differ. You can try to negate the logic check.

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

Reply:
still valid after your recent changes

Reply:
The misunderstanding is that Tumbleweed is not a version, it's a product. Also following https://opensource.suse.com/legal/policy Factory and hence Tumbleweed is gaining new features first before downstream products like Leap and SLE get a feature. This means that code should be structured to treat Tumbleweed the default, not mention it at all and handle all different behaviour differently. So instead of

```
if (!tumbleweed) {
    handle_old_behaviour();
}
else {
    handle_new_behaviour();
}
```

one should write

```
if (!is_sle('<=old') && !is_leap('<=old')) {
    handle_new_behaviour();
}
else {
    handle_old_behaviour();
}
```

with "old" being the newest downstream product and version that is still relying on the old behaviour.